docs/migrations.rst

```diff
@@ -0,0 +1,179 @@
+   2. If needed, access the ``url`` using ``self.url``
+
+Global plugin arguments
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Streamlink 5.3.0 :ref:`deprecated <deprecations:Global plugin arguments>` the ``is_global=True`` argument
+of the :py:meth:`pluginargument <streamlink.plugin.pluginargument>` decorator (as well as the
+:py:class:`Argument <streamlink.options.Argument>` class), as global plugin arguments were deemed unnecessary.
+The ``is_global`` argument has thus been removed now.
+
+| :octicon:`x-circle` #5140
+| :octicon:`git-pull-request` #5033
+
+.. admonition:: Migration
+   :class: hint
+
+   1. Get the value of the global argument using :py:meth:`Streamlink.get_option() <streamlink.session.Streamlink.get_option>`
+      instead of getting it from :py:attr:`Plugin.options <streamlink.plugin.Plugin.options>`
+
+plugin.api.validate.text
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Streamlink 5.2.0 :ref:`deprecated <deprecations:plugin.api.validate.text>` the ``plugin.api.validate.text`` alias for ``str``.
+This was a remnant of the Python 2 era and has been removed.
+
+| :octicon:`x-circle` #5090
+| :octicon:`git-pull-request` #5404
+
+.. admonition:: Migration
+   :class: hint
+
+   1. Replace ``plugin.api.validate.text`` with ``str``
+
+HTTPStream and HLSStream signature changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The signatures of the constructors of :py:class:`HTTPStream <streamlink.stream.HTTPStream>`
+and :py:class:`HLSStream <streamlink.stream.HLSStream>`, as well as
+the :py:meth:`HLSStream.parse_variant_playlist() <streamlink.stream.HLSStream.parse_variant_playlist>` classmethod
+got changed and fixed.
+
+| :octicon:`git-pull-request` #5429
+
+.. admonition:: Migration
+   :class: hint
+
+   1. Set the :py:class:`Streamlink <streamlink.session.Streamlink>` session instance as a positional argument,
+      or replace the ``session_`` keyword with ``session``
+
+
+streamlink 5.0.0
+----------------
+
+Session.resolve_url() return type changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+With the removal of the ``Plugin.bind()`` classmethod, the return value of
+:py:meth:`Streamlink.resolve_url() <streamlink.session.Streamlink.resolve_url>`
+and :py:meth:`Streamlink.resolve_url_no_redirect() <streamlink.session.Streamlink.resolve_url_no_redirect>`
+got changed. Both methods now return a three-element tuple of the resolved plugin name, plugin class and URL.
+
+| :octicon:`git-pull-request` #4768
+
+.. admonition:: Migration
+   :class: hint
+
+   1. Return type changed from ``tuple[type[Plugin], str]`` to ``tuple[str, type[Plugin], str]``
+
+
+streamlink 4.0.0
+----------------
+
+streamlink.plugin.api.utils
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``streamlink.plugin.api.utils`` module including the ``itertags`` function and export aliases for ``streamlink.utils.parse``
+has been removed. Import the parse functions directly and find data in XML/HTML by parsing it via ``parse_{xml,html}`` and
```

Comment:
`has been removed` -> `have been removed`.

Reply:
"Removed" was about the (singular) module, not its contents. Reformatted the section, because it was redundant anyway.